apps/framework-cli/src/utilities/package_managers.rs

```diff
@@ -148,6 +148,42 @@ pub fn link_sdk(
     Ok(())
 }
 
+/// Detects the package manager to use based on lock files present in the project directory.
+///
+/// The detection follows this priority order:
+/// 1. pnpm-lock.yaml -> pnpm
+/// 2. package-lock.json -> npm
+/// 3. Default to npm if no lock files found
+///
+/// # Arguments
+///
+/// * `project_dir` - Path to the project directory to scan for lock files
+///
+/// # Returns
+///
+/// * `PackageManager` - The detected package manager
+pub fn detect_package_manager(project_dir: &PathBuf) -> PackageManager {
+    use crate::utilities::constants::{PACKAGE_LOCK_JSON, PNPM_LOCK};
+
+    debug!("Detecting package manager in directory: {:?}", project_dir);
+
+    // Check for pnpm-lock.yaml first (most specific)
+    if project_dir.join(PNPM_LOCK).exists() {
```

Comment:
I believe you also need to look up the tree for `pnpm` the lock file might not be in the project directory but in one of the parent